File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.kt
class Solution {
    fun maxProfit(prices: IntArray): Int {
        var answer = 0
        var curr = prices[0]

        for (i in 1 until prices.size){
            when {
                curr > prices[i]  -> curr = prices[i]
                else -> answer = max(answer, (prices[i] - curr))
            }
        }
        return answer
    }
}

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        curr = prices[0]
        answer = 0
        for i in range(1, len(prices)):
            if prices[i] > curr:
                answer = max(answer, prices[i] - curr)
            else:
                curr = prices[i]
        return answer